lib/floki/html_tree.ex

```diff
@@ -99,14 +99,18 @@ defmodule Floki.HTMLTree do
   defp build_tree(tree, [], _, []), do: tree
   defp build_tree(tree, [{tag, attrs, child_children} | children], parent_id, stack) do
     new_id = IDSeeder.seed(tree.node_ids)
-    new_node = %HTMLNode{type: tag,
-                         attributes: attrs,
-                         node_id: new_id,
-                         parent_node_id: parent_id}
+    nodes = if tag != :pi do
```

Comment:
I think we could ignore the nodes that are `:pi` in a function match.
So, above this function declaration, we could have a declaration more or less like this:

```elixir
defp build_tree(tree, [{:pi, _, _} | children], parent_id, stack), do: build_tree(tree, children, parent_id, stack)
```

This version would avoid an `if`. What do you think?